ingest/mtx.py

```diff
@@ -28,116 +33,134 @@
     from .monitor import trace
 
 
-class Mtx(GeneExpression):
-    ALLOWED_FILE_TYPES = ["text/csv", "text/plain", "text/tab-separated-values"]
+class MTXIngestor(GeneExpression):
+    ALLOWED_FILE_TYPES = ["text/tab-separated-values"]
+
+    def matches_file_type(file_type):
+        return 'mtx' == file_type
 
     def __init__(self, mtx_path: str, study_file_id: str, study_id: str, **kwargs):
-        self.tracer = kwargs.pop("tracer")
         GeneExpression.__init__(self, mtx_path, study_file_id, study_id)
 
         genes_path = kwargs.pop("gene_file")
         genes_ingest_file = IngestFiles(genes_path, self.ALLOWED_FILE_TYPES)
-        self.genes_file, genes_local_path = genes_ingest_file.resolve_path(genes_path)
+        self.genes_file, genes_local_path = genes_ingest_file.resolve_path(
+            genes_path)
 
         barcodes_path = kwargs.pop("barcode_file")
-        barcodes_ingest_file = IngestFiles(barcodes_path, self.ALLOWED_FILE_TYPES)
+        barcodes_ingest_file = IngestFiles(
+            barcodes_path, self.ALLOWED_FILE_TYPES)
         self.barcodes_file, barcodes_local_path = barcodes_ingest_file.resolve_path(
             barcodes_path
         )
 
         mtx_ingest_file = IngestFiles(mtx_path, self.ALLOWED_FILE_TYPES)
-        mtx_file, self.mtx_local_path = mtx_ingest_file.resolve_path(mtx_path)
+        self.mtx_file, self.mtx_local_path = mtx_ingest_file.resolve_path(
+            mtx_path)
+        # A list ['N', 'K', 'M'] that represents a gene-barcode matrix where N
+        # is the gene index, M is the barcode index, and K is the expresion score
+        # for the given gene index
+        self.mtx_description = linecache.getline(
```

Comment:
Nifty, [`linecache` looks fast](https://stackoverflow.com/a/19190196) -- thanks for putting it on my radar.